pandas/tests/arrays/floating/test_comparison.py

```diff
@@ -34,3 +36,27 @@ def test_equals():
     a1 = pd.array([1, 2, None], dtype="Float64")
     a2 = pd.array([1, 2, None], dtype="Float32")
     assert a1.equals(a2) is False
+
+
+def test_equals_nan_vs_na():
+    # GH#44382
+
+    mask = np.zeros(3, dtype=bool)
+    data = np.array([1.0, np.nan, 3.0], dtype=np.float64)
+
+    left = FloatingArray(data, mask)
+    assert left.equals(left)
```

Comment:
Could you do something like `left2 = FloatingArray(data.copy(), mask.copy())` to create the other array to test equality with? I would just avoid the special case of "identical" objects (in case we might add a fastpath for that in the future)

Reply:
updated; is this what you had in mind?

Reply:
That's fine! 
Can you add a small whatsnew note?